core/common/src/main/java/alluxio/security/authentication/AbstractSaslServerHandler.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.security.authentication;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.security.sasl.SaslException;
+import javax.security.sasl.SaslServer;
+
+/**
+ * Abstract {@link SaslServerHandler} implementation that maintains {@link SaslServer} instance.
+ */
+public abstract class AbstractSaslServerHandler implements SaslServerHandler {
```

Comment:
These classes looks identical in logic. Is there a way to further unify the code path?

Reply:
You mean dropping `SaslServerHandler` maybe?

Reply:
not sure about that, I was just thinking of having something like a `AbstractSaslHandler<SaslServer>` so we don't have to repeat the logic in two classes.

Reply:
`SaslClient` and `SaslServer` are not sharing an ancestor even though their logic looks identical. That makes it hard to unify their abstract handlers without lots of boilerplate...